test/testing_utils/checkers.go

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright (C) 2017 XLAB, Ltd.
+ *
+ * This work is open source software, licensed under the terms of the
+ * BSD license as described in the LICENSE file in the top-level directory.
+ */
+
+package test
+
+import (
+	"archive/tar"
+	"compress/gzip"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	"reflect"
+
+	. "gopkg.in/check.v1"
+)
+
+// TarGzEquals checker checks that given tar.gz archive contains exactly given files
+// with given content.
+//
+// For example:
+//
+//     c.Assert("/tmp/archive.tar.gz", TarGzEquals, map[string]string{"/file01.txt": "Exact content"})
+//
+var TarGzEquals Checker = &tarGzEqualsChecker{
+	&CheckerInfo{Name: "TarGzEquals", Params: []string{"obtained", "expected"}},
+}
+
+type tarGzEqualsChecker struct {
+	*CheckerInfo
+}
+
+func (checker *tarGzEqualsChecker) Check(params []interface{}, names []string) (result bool, error string) {
+	defer func() {
+		if v := recover(); v != nil {
+			result = false
+			error = fmt.Sprint(v)
+		}
+	}()
+
+	tarGzPath, ok := params[0].(string)
+	if !ok {
+		return false, "ERROR: `obtained` must be a path to tar.gz file"
```

Comment:
Looking at various gocheck checkers I suggest two changes to these error messages

* do not prefix with `ERROR:`
* use simply `Obtained` or `Obtained value`. Using back ticks in error messages does not make sense as it'll be printed in the console